tests/api_test/api_test.cpp

```diff
@@ -547,6 +547,30 @@ _test_nested_maps(bpf_map_type type)
 TEST_CASE("array_map_of_maps", "[map_in_map]") { _test_nested_maps(BPF_MAP_TYPE_ARRAY_OF_MAPS); }
 TEST_CASE("hash_map_of_maps", "[map_in_map]") { _test_nested_maps(BPF_MAP_TYPE_HASH_OF_MAPS); }
 
+TEST_CASE("duplicate_fd", "")
+{
+    fd_t map_fd1 = bpf_map_create(BPF_MAP_TYPE_ARRAY, "map", sizeof(uint32_t), sizeof(uint32_t), 1, nullptr);
+    REQUIRE(map_fd1 > 0);
+
+    uint32_t key = 0;
+    uint32_t value = 1;
+    REQUIRE(bpf_map_update_elem(map_fd1, &key, &value, 0) == 0);
+
+    fd_t map_fd2;
+    REQUIRE(ebpf_duplicate_fd(map_fd1, &map_fd2) == EBPF_SUCCESS);
+    REQUIRE(map_fd2 > 0);
+
+    REQUIRE(bpf_map_lookup_elem(map_fd2, &key, &value) == 0);
+    REQUIRE(value == 1);
+
+    REQUIRE(ebpf_close_fd(map_fd2) == EBPF_SUCCESS);
+    REQUIRE(ebpf_close_fd(map_fd2) == EBPF_FAILED);
```

Comment:
This is failing. It actually failed earlier as well, but a scripting error masked the failure. It has been fixed since. Please see the comment in platform.cpp.

Reply:
I must-have misread your previous code change. You are deliberately trying to close the same fd twice. You have disabled the CRT handler. @Alan-Jowett please see if there is any alternative way.
Meanwhile I am approving this PR.